src/odemis/acq/align/goffset_alignment.py

```
@@ -0,0 +1,200 @@
+
+    runs = n_detectors + max(0, n_gratings - 1)
+    move_time = ((n_gratings-1) * MOVE_TIME_GRATING + (n_detectors - 1) * MOVE_TIME_DETECTOR)
+
+    # total time = time spent running alignment algorithms + time spent moving hardware
+    return runs * EST_ALIGN_TIME + move_time
+
+
+def auto_align_grating_detector_offsets(spectrograph: model.Actuator,
+                                    detectors: Union[model.Detector, List[model.Detector]],
+                                    selector: Optional[model.Actuator] = None,
+                                    streams: Optional[List['Stream']] = None) -> model.ProgressiveFuture:
```

Ruff (0.15.4)

[error] Undefined name `Stream`

(F821)